source/dialogs/drun.c

```diff
@@ -547,6 +556,60 @@ static void drun_icon_fetch ( gpointer data, gpointer user_data )
     rofi_view_reload ();
 }
 
+/**
+ * Initialize desktop entry fields.
+ */
+static void drun_collect_entry_fields ( DRunModePrivateData *pd )
+{
+    unsigned int entry_fields_length = 4;
+    pd->matching_entry_fields                     = g_realloc ( pd->matching_entry_fields, entry_fields_length * sizeof ( DRunEntryField ) );
+    pd->matching_entry_fields[0].entry_field_name = "name";
+    pd->matching_entry_fields[0].enabled          = TRUE;
+    pd->matching_entry_fields[1].entry_field_name = "generic";
+    pd->matching_entry_fields[1].enabled          = TRUE;
+    pd->matching_entry_fields[2].entry_field_name = "exec";
+    pd->matching_entry_fields[2].enabled          = TRUE;
+    pd->matching_entry_fields[3].entry_field_name = "categories";
+    pd->matching_entry_fields[3].enabled          = TRUE;
+    pd->matching_entry_fields_length              = 4;
+}
+
+static void drun_mode_parse_entry_fields ( DRunModePrivateData *pd )
+{
+    char               *savept = NULL;
+    // Make a copy, as strtok will modify it.
+    char               *switcher_str = g_strdup ( config.drun_match_fields );
+    const char * const sep           = ",#";
+    // Split token on ','. This modifies switcher_str.
+    for ( unsigned int i = 0; i < pd->matching_entry_fields_length; i++ ) {
+        pd->matching_entry_fields[i].enabled = FALSE;
+    }
+    for ( char *token = strtok_r ( switcher_str, sep, &savept ); token != NULL;
+          token = strtok_r ( NULL, sep, &savept ) ) {
+        if ( strcmp ( token, "all" ) == 0 ) {
+            for ( unsigned int i = 0; i < pd->matching_entry_fields_length; i++ ) {
+                pd->matching_entry_fields[i].enabled = TRUE;
+            }
+            break;
+        }
+        else {
+            gboolean matched = FALSE;
+            for ( unsigned int i = 0; i < pd->matching_entry_fields_length; i++ ) {
+                const char * entry_name = pd->matching_entry_fields[i].entry_field_name;
+                if ( strcmp ( token, entry_name ) == 0 ) {
+                    pd->matching_entry_fields[i].enabled = TRUE;
+                    matched                              = TRUE;
+                }
+            }
+            if ( !matched ) {
+                g_warning ( "Invalid entry name :%s", token );
```

Comment:
Not sure yet myself? should we make this a 'stronger' error and display it to the user?

Reply:
I don't know what your policy is for this kind of problems. I got inspiration from the parsing function for the combi-modi option, so it would be coherent to keep the same type of warning in all parsing functions.